src/docs/arch-snapshot.dox

```diff
@@ -1,9 +1,169 @@
+@startuml{snapshot.png}
+digraph g {
+graph [
+ordering="in";
+rankdir = "LR";
+];
+node [
+fontsize = "16"
+];
+edge [
+];
+"node1" [
+label = "<f0> Snapshot [TXID 8] | <f1> Concurrent Transactions | <f2> Maximum Transaction ID | <f3> Minimum Transaction ID"
+shape = "record"
+];
+"node0" [
+label = "<f0> Global Transaction State | <f1> Running Transactions| <f2> Current Transaction ID"
+shape = "record"
+];
+"node2" [
+label = "<f0> 3"
+shape = "record"
+];
+"node3" [
+label = "<f0> 3"
+shape = "record"
+];
+"node4" [
+label = "<f0> 5"
+shape = "record"
+];
+"node6" [
+label = "<f0> 7"
+shape = "record"
+];
+"node7" [
+label = "<f0> 8"
+shape = "record"
+];
+"node8" [
+label = "<f0> 5"
+shape = "record"
+];
+"node10" [
+label = "<f0> 7"
+shape = "record"
+];
+"node11" [
+label = "<f0> 9"
+shape = "record"
+];
+"node12" [
+label = "<f0> 3"
+shape = "record"
+];
+"node13" [
+label = "<f0> 9"
+shape = "record"
+];
+"node1":f1 -> "node2":f0 [
+id = 0
+];
+"node2":f0 -> "node8":f0 [
+id = 0
+];
+"node8":f0 -> "node10":f0 [
+id = 0
+];
+"node0":f1 -> "node3":f0 [
+id = 0
+];
+"node3":f0 -> "node4":f0 [
+id = 0
+];
+"node4":f0 -> "node6":f0 [
+id = 0
+];
+"node6":f0 -> "node7":f0 [
+id = 0
+];
+"node0":f2 -> "node11":f0 [
+id = 0
+];
+"node1":f2 -> "node13":f0 [
+id = 0
+];
+"node1":f3 -> "node12":f0 [
+id = 0
+];
+}
+@enduml
+@plantuml_end
+
+@section snapshot_visibility Snapshot Visibility
+
+When determining if a transaction can read a given record, WiredTiger applies a visibility function to discern whether the committed value is
+visible. When using \c snapshot and \c read-committed isolated transactions, WiredTiger's visibility function will leverage a transactions snapshot information
+to determine if a given record is visible. When determining if a given transaction ID is visible to a snapshot,
+WiredTiger applies the below checks in the following order:
+1. The transaction ID is greater than or equal to the snapshots recorded maximum transaction ID: If satisfied, the transaction ID is invisible, as the ID
+would've been allocated after the snapshot was created.
+2. The transaction ID is less than the snapshots recorded minimum transaction ID: If satisfied, the transaction ID is visible, as these would've been
+allocated & committed before the snapshot was created.
+3. The transaction ID exists in the snapshots recorded concurrent transaction list: If found, the transaction ID is invisible, as this ID wasn't committed at the
+time the snapshot was created.
+4. Otherwise the ID is visible.
+
+The early ordering of checking whether the transaction ID sits outside the maximum and minimum bounds is an important detail since the concurrent transaction list
+can be empty i.e. snapshot was taken a time when no concurrent transactions where running.
+
+@section snapshot_checkpoint Snapshots \& Checkpoints
+
+Checkpoints require a snapshot to write a transactionally consistent view of the data. A checkpoint maintains it own special global
```

Comment:
"A checkpoint maintains it own special global" --> "A checkpoint maintains its own special global"
i.e.
"it" --> "its"

Reply:
Fixed! 551a4e12a694b00d764c94568c17c2cb8bce6f22